libbeat/template/template.go

```diff
@@ -69,10 +69,15 @@ func New(beatVersion string, beatName string, esVersion common.Version, config T
 
 	event := &beat.Event{
 		Fields: common.MapStr{
+			// beat object was left in for backward compatibility reason for older configs.
 			"beat": common.MapStr{
 				"name":    beatName,
 				"version": bV.String(),
 			},
+			"agent": common.MapStr{
+				"name":    beatName,
+				"version": bV.String(),
+			},
```

Comment:
As just discussed, the `apm-server` is using `observer` instead of `agent`, so the `observer` also needs to be added here. 

Moving away from `beat` to `agent` causes some issues in APM as an `agent` means something different there. Therfore, hardcoding to use `beatName` as `agent.name` and same for version information is more a quickfix than a nice solution from my perspective.
I think we should come up with a more reliable solution for this in the long run, e.g. the possibility to define something like a `beatId` globally, which is set to `agent` by default, but can be overwritten from every beat.

Reply:
Should I add observer also as part of this PR to have a quick fix?

Reply:
yes please, it is a precondition to fix https://github.com/elastic/apm-server/issues/1610, as splitting indices is broken since `beat` changed to `agent`.

Reply:
added